crypto/x509/x509_cmp.c

```diff
@@ -208,7 +208,7 @@ int X509_add_cert(STACK_OF(X509) *sk, X509 *cert, int flags)
                 return 1;
         }
     }
-    if ((flags & X509_ADD_FLAG_NO_SS) != 0 && X509_self_signed(cert, 0))
+    if ((flags & X509_ADD_FLAG_NO_SS) != 0 && X509_self_signed(cert, 0) > 0)
         return 1;
```

Comment:
You probably want to return 0 in case X509_self_signed returns negative

Reply:
Do you mean like this: 
```
if ((flags & X509_ADD_FLAG_NO_SS) != 0) {
    if (X509_self_signed(cert, 0) > 0) 
         return 1;
    else:
         return 0;
}
```

Reply:
yes

Reply:
Something like::
```c
if ((flags & X509_ADD_FLAG_NO_SS) != 0) {
    int ret = X509_self_signed(cert, 0);
    if (ret != 0)
         return ret > 0 ? 1 : 0;
}
```